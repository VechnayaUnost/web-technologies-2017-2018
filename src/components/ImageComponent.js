import React from 'react';

function ImageComponent(props) {
    if (props.icon != null) {
        return (
            <div className="BioComponent">
                <img alt={props.userSocial} src={props.icon}></img>
            </div>
        );
    }else{
        return(<div/>);
    }
}



export default ImageComponent;