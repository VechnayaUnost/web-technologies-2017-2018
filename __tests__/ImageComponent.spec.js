import React from 'react';
import ImageComponent from '../src/components/ImageComponent';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson, mountToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('ImageComponent component test', () => {
    it('ImageComponent component rendered succesfully', () => {
        const imageComponent = mount(
            <ImageComponent props={{ userSocial: '@neyney', icon: 'twitter' }} />
        );
        expect(mountToJson(imageComponent)).toMatchSnapshot();
    });

    it('ImageComponent component rendered with null icon props', () => {
        const imageComponent = mount(
            <ImageComponent props={{ userSocial: '@neyney', icon: null }} />
        );
        expect(imageComponent.find('img').length).toBe(0);
    });
});