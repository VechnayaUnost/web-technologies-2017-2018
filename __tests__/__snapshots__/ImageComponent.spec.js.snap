// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageComponent component test ImageComponent component rendered succesfully 1`] = `
<ImageComponent
  props={
    Object {
      "icon": "twitter",
      "userSocial": "@neyney",
    }
  }
>
  <div />
</ImageComponent>
`;
